src/modules/insights/components/Filter.tsx

```diff
@@ -42,16 +43,20 @@ class Filter extends React.Component<Props, States> {
     };
   }
 
-  onTypeChange = (value: any) => {
-    const integrationType = value ? value.value : '';
-    this.setState({ integrationType });
-    router.setParams(this.props.history, { integrationType });
+  onTypeChange = (integrations: any) => {
+    this.setState({ integrationIds: integrations.map(el => el.value) });
   };
 
-  onBrandChange = (value: any) => {
-    const brandId = value ? value.value : '';
-    this.setState({ brandId });
-    router.setParams(this.props.history, { brandId });
+  onBrandChange = (brands: any) => {
+    this.setState({ brandIds: brands.map(el => el.value) });
+  };
+  onApplyClick = () => {
```

Comment:
1. Space

2. const { history } = this.props;
    const { integrationIds, brandIds } = this.state;

    router.setParams(history, {
      integrationIds: (integrationIds || []).join(','),
      brandId: (brandIds || []).join(',')
    });

Reply:
done